Migration to application commands

Joosemi02 opened the pull request:
Replace old message commands for slash commands

https://support-dev.discord.com/hc/en-us/articles/6025578854295-Why-We-Moved-to-Slash-Commands

Joosemi02 committed:
Rename discord module (mirror) to discord.py

Joosemi02 committed:
Minor spelling changes

Joosemi02 committed:
Code cleanup

- Remove unused imports
- Group imports at the top of the file as per PEP8
- Change prefix to bot @mention so message intent isn't required
- Remove on't assignment to builtin variable "str"
- Minor format changes
 - Change ' to "
 - Fix spacing

Joosemi02 committed:
Migration to application commands

- Add sync command to sync bot tree application commands
- Migrate text commands to slash application commands

andalugeek commented:
Under review

Joosemi02 committed:
Add how to sync in README.md

fontanon commented:
Hi @Joosemi02 thanks for the PR. Unfortunately I'm not able to make it work.

In order to make the bot run, I had to edit `app/andaluhbot.py` as per the following image:

![image](https://github.com/andalugeeks/andaluh-discord/assets/113024/34674732-d068-4104-aeb8-f8ed4e0c7c75)

But then after having the bot successfully runing on my test server ...

```
$ python3 app/andaluhbot.py 
2023-11-08 08:42:07 INFO     discord.client logging in using static token
2023-11-08 08:42:08 INFO     discord.gateway Shard ID None has connected to Gateway (Session ID: f2099ef6068a100c5f55f703034cbf55).
Logged in as <discord.ext.commands.bot.Bot object at 0x7f5cbfd8fa90>
```

... no command produces a response:

![image](https://github.com/andalugeeks/andaluh-discord/assets/113024/3b7a29e5-ae71-4bfc-9dfe-62c94d40b526)

You may want to join AndaluGeeks developers community at discord to discuss if that helps: https://discord.gg/bWGNHw3

Joosemi02 committed:
Bug fixes

- Change "help" parameter in @bot.tree.command for "description"
- Fix duplicate command name

Joosemi02 committed:
Merge branch 'master' of https://github.com/Joosemi02/andaluh-discord

Joosemi02 committed:
- Change intents to default
- Format

Joosemi02 committed:
Defer interactions responses

Joosemi02 commented:
Hello @fontanon, thanks for your PR review.

I fixed the bugs you mentioned in the following commit: 9d347783118440cc3dddbc75caa8875899e73d81.

Then, after the bot is successfully running, you'll need to use the sync command for the slash commands to show up.
Using sync: `@andaluh_test_fontanon sync`.

After a few seconds/minutes the slash commands show show up in a list when typing `/`.
Using slash commands: 
<img width="520" alt="image" src="https://github.com/andalugeeks/andaluh-discord/assets/37875402/8784c61b-2ac8-4fbd-a790-acad8e1c08b8">.

Also note the bot needs to be added with the applications.commands scope:
<img width="643" alt="image" src="https://github.com/andalugeeks/andaluh-discord/assets/37875402/118a9363-70fb-4c14-aeb1-c1ccb8d2900c">
